GamepadEventDrivenInputAPI/explainer.md

```diff
@@ -0,0 +1,237 @@
+
+Developers working on latency-sensitive applications, such as cloud gaming platforms, have reported needing to poll at very high frequencies (e.g., every 4 ms) to detect input as quickly as possible. However, even with aggressive polling, scripts may still struggle to react in real time, especially under heavy UI thread load or on resource-constrained devices.
+
+An event-driven Gamepad API (similar to existing keyboard and mouse event models) would allow applications to respond immediately to input changes as they occur, reducing the reliance on polling and enabling real-time responsiveness for latency-critical use cases.
+
+### Developer code sample of existing poll based API
+```JS
+function pollGamepadInput() {
+  const gamepads = navigator.getGamepads();
+
+  for (const gamepad of gamepads) {
+    if (!gamepad) continue;
+    // Example: Logging the first axis and button.
+    const axisX = gamepad.axes[0];
+    const buttonA = gamepad.buttons[0].pressed;
+
+    console.log(`Axis X: ${axisX}, Button A pressed: ${buttonA}`);
+  }
+
+  // Continue polling in the next animation frame.
+  requestAnimationFrame(pollGamepadInput);
+}
+
+// Start polling.
+window.addEventListener('gamepadconnected', () => {
+  console.log('Gamepad connected!');
+  requestAnimationFrame(pollGamepadInput);
+}); 
+```
+#### Key Points:
+- navigator.getGamepads() returns a snapshot of all connected gamepads.
+- The polling loop is driven by `requestAnimationFrame`, typically around 60Hz (matching display refresh rate), which is much lower than the internal OS poll rate (eg., 250Hz). This mismatch can result in missed input updates, making the 60Hz rate insufficient for latency-critical applications like cloud gaming.
+
+## Goals
+
+Reduce input latency by moving away from constant polling and introducing event-driven input handling. 
+
+## Non-goals
+
+The existing polling mechanism will not be deprecated. We are just proposing an alternative way of handling input events and applications are free to select whichever they prefer.
+
+## Proposed Approach
+To address the challenges of input latency, this proposal introduces a new event-driven mechanism: The `rawgamepadinputchange` event. This event fires directly on the [Gamepad](https://w3c.github.io/gamepad/#dom-gamepad) object and delivers real-time updates for each input frame, eliminating the need for high-frequency polling. The `rawgamepadinputchange` event includes detailed information about the state of the gamepad at the moment of change:
```

Comment:
nit: sorry I missed this previously, but the original proposal named this event `rawgamepadchange` instead of `rawgamepadinputchange`.  Should we use the shorter name as well?